src/main/java/com/minecolonies/core/datalistener/RecruitmentItemsListener.java

```diff
@@ -95,8 +98,13 @@ else if (level > IRON_SKILL_LEVEL)
         {
             minimumRarity = 0;
         }
-        final List<RecruitCost> recruitCostStream = RECRUIT_COSTS.stream().filter(f -> f.rarity <= minimumRarity).toList();
-        return recruitCostStream.get(source.nextInt(recruitCostStream.size()));
+        final List<RecruitCost> recruitCosts = RECRUIT_COSTS.stream().filter(f -> f.rarity >= minimumRarity).toList();
+        if (recruitCosts.isEmpty())
+        {
+            Log.getLogger().warn("No recruitment items found, please ensure valid recruitment items exist in the datapacks.");
```

Comment:
This should be during loading(#apply) :) So you know which datapack file and which part of the datapack is wrong, and what you need to add/change in the pack

Reply:
It doesn't necessarily have to be during load, if one does add a single low rarity item and has no higher rarity items, it can result in no items in the filter if a higher level citizen has to spawn, hence why I put it here instead

Reply:
but the list does not change after loading, so can check the full list once and warn once about it

Reply:
But then still, if I have no data on load, then what? It still can be null

Reply:
I'm not telling you to remove the null check, I'm telling you to on data load verify the items loaded and log an error if there is a problem within